### PR TITLE
Backend updates coming to main branch.

### PR DESCRIPTION
**Pull request by JuliusJauga**
Collaborated with Edvinas and Aurelijus. Added backend logic for mode 1 and user registration. Remote database is up. Changes in docker, database schemas are made, API endpoints done, Unit tests for user data handling, still need to write for task 1.

### CODE REVIEW

**Comment by JuliusJauga** [comment]
Comments need to be deleted from login and register pages!

File: server/server/connect_postgres.sh
Comment (AurelijusLuksas) [inline server/server/connect_postgres.sh:1]: This file has to be deleted, because the enviroment variables are not set.

**Review by AurelijusLuksas — CHANGES_REQUESTED** [review]
Looks pretty good, but someone else should take a look too.

File: server/server/config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (tikis23) [inline server/server/config.json:1]: Config should be in a secret env

**Review by RokasBal — COMMENTED** [review]
Everything seems to be good

**Comment by JuliusJauga** [comment]
Also during development you have to create .env file in the parent folder with the required field.

File: server/server/connect_postgres.sh
Comment (JuliusJauga) [reply server/server/connect_postgres.sh:1]: Deleted the file.

File: server/server/src/Program.cs [hunk 2]
```diff
@@ -1,37 +1,45 @@
...
 
-
-            // Add services to the container.
-            // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
-            builder.Services.AddDbContext<FlashDbContext>(options => 
-                options.UseSqlServer("Server=localhost\\SQLEXPRESS01;Database=flash-read-db;Trusted_Connection=True;"));
             
+            try
+            {
+                var dataSourceBuilder = new NpgsqlDataSourceBuilder(ConnectionStringBuilder.BuildConnectionString());
```
Comment (tikis23) [inline server/server/src/Program.cs:25]: We should handle the case when we fail to connect to the database. The program can exit or run a loop trying to connect again.

File: server/server/config.json (hunk 1 above)
Comment (JuliusJauga) [reply server/server/config.json]: Made the changes.

**Review by tikis23 — APPROVED** [review]

File: server/server/src/Program.cs (hunk 2 above)
Comment (JuliusJauga) [reply server/server/src/Program.cs:25]: On it.

**Pull request merged**